R/auth_sitrep.R

```diff
@@ -0,0 +1,187 @@
+  path <- tools::R_user_dir("rtweet", "config")
+  if (!dir.exists(path)) {
+    dir.create(path, showWarnings = FALSE, recursive = TRUE)
+  }
+  
+  if (is.null(all_tokens_files)) {
+    inform("No tokens where found! See auth_as help file.")
+    return(NULL)
+  }
+  
+  type_auth <- type_auth(all_tokens_files)
+  all_tokens <- lapply(all_tokens_files, readRDS)
+  names(all_tokens) <- all_tokens_files
+  
+  if (any(type_auth == "bearer")){
+    handle_bearer(all_tokens[type_auth == "bearer"], path)
+  }
+  if (any(type_auth == "token")){
+    handle_token(all_tokens[type_auth == "token"], path)
+  }
+  invisible(list.files(path, pattern = "*.rds", all.files = TRUE, 
+                       full.names = TRUE))
+}
+
+find_old_tokens <- function() {
+  twitter_pat <- Sys.getenv("TWITTER_PAT")
+  home_path <- normalizePath(file.path("~"), mustWork = TRUE)
+  
+  many_paths <- c(twitter_pat, home_path)
+  old_tokens <- lapply(many_paths, list.files, pattern = ".rtweet_token.*rds", 
+                       full.names = TRUE, all.files = TRUE)
+  unlist(old_tokens, TRUE, FALSE)
+}
+
+find_rappdirs_tokens <- function() {
+  if (requireNamespace("rappdirs", quietly = TRUE)) {
+    rappdirs_path <- getOption("rtweet:::config_dir", 
+                               rappdirs::user_config_dir("rtweet"))
+    rappdirs_tokens <- list.files(rappdirs_path, pattern = "*.rds", 
+                                  all.files = TRUE, full.names = TRUE)
+  } else {
+    rappdirs_tokens <- NULL
+  }
+  rappdirs_tokens
+}
+
+find_tools_tokens <- function() {
+  path <- tools::R_user_dir("rtweet", "config")
+  final_path <- getOption("rtweet:::config_dir", path)
+  if (dir.exists(final_path)) {
+    new_tokens <- list.files(final_path, pattern = "*.rds", 
+                             all.files = TRUE, full.names = TRUE)
+  } else {
+    new_tokens <- NULL
+  }
+}
+
+bearer_auth <- function(bearer) {
+  tok <- vapply(bearer, function(x){x$token}, character(1L))
+  tok <- as.factor(tok)
+  levels(tok) <- LETTERS[length(unique(tok))]
+  df <- data.frame(token = tok)
+  rownames(df) <- bearer
+}
+
+token_auth <- function(tokens) {
+  tokens <- lapply(tokens, raw_auth)
+  n <- length(tokens)
+  df <- data.frame(app = character(n), 
+                   user_id = character(n), 
+                   key = character(n))
+  for (i in seq_along(tokens)) {
+    token <- tokens[[i]]
+    df[i, names(token)] <- token
+  }
+  rownames(df) <- names(tokens)
+  uk <- unique(df$key)
+  length_levels <- length(uk) - sum(any(uk == ""))
+  df$key <- factor(df$key, labels = LETTERS[seq_len(length_levels)], exclude = "")
+  df
+}
+
+#' @importFrom methods is
+type_auth <- function(old_tokens_files) {
+  old_tokens <- lapply(old_tokens_files, readRDS)
+  names(old_tokens) <- old_tokens_files
+  class_tokens <- sapply(old_tokens, is)
+  class_tokens <- ifelse(endsWith(class_tokens, "Token1.0"), "token", "bearer")
+}
+
+move_tokens <- function(tokens, folder) {
```

Comment:
I worry that you have a lot of complicated code here and no tests. Tests are particularly important for base R file manipulating code, since most base R functions do not error if the operation fails (instead they warn and return a logical vector indicating whether or not each operation succeeded or failed).